backend/internal/adapters/primary/handlers/internal/gemservice.go

```diff
@@ -43,6 +44,32 @@ func (s *gemService) Get(c context.Context, r *gemsapi.GetRequest) (*gemsapi.Get
 	}, nil
 }
 
+func (s *gemService) GetPendingCollectionByUser(c context.Context, r *gemsapi.GetPendingCollectionByUserRequest) (*gemsapi.GetPendingCollectionByUserResponse, error) {
+	gems, err := s.usecases.GetPendingCollectionByUser(r.UserId)
+	if err != nil {
+		return nil, err
+	}
+
+	processedGems := []*gemsapi.Gem{}
+
+	for _, gem := range gems {
+		processedGem := gemsapi.Gem{
+			Id:         gem.ID,
+			Message:    gem.Message,
+			Latitude:   gem.Latitude,
+			Longitude:  gem.Longitude,
+			CreatorId:  gem.CreatorId,
+			CreatedAt:  &timestamppb.Timestamp{Seconds: gem.CreatedAt.Unix()},
+			ReceiverId: gem.ReceiverId,
```

Comment:
> Is there a nicer way to handle nil fields, other than checking them individually?

We can either make use of nullable types (as mentioned in one of the other comments) but have the drawback of having to account for `nil` everywhere, otherwise we just need to handle `nil` here. There are some helper functions in the `timestamppb` package though such as `IsValid` and `AsTime`.

> Also, I don't suppose Golang has a way to "spread" a struct right

I think you can try embedding the struct if it is a subset of the other struct.

Reply:
> I think you can try embedding the struct if it is a subset of the other struct.

Hmm, you mean using composition right? I don't think this works since `gemsapi.Gem` is auto-generated by protobuf